packages/formik/src/Formik.tsx

```diff
@@ -504,35 +510,37 @@ export function useFormik<Values extends FormikValues = FormikValues>({
             value: maybePromise as string | undefined,
           },
         });
-        return Promise.resolve(maybePromise as string | undefined);
+        return Promise.resolve(maybePromise);
       }
     } else if (props.validationSchema) {
       dispatch({ type: 'SET_ISVALIDATING', payload: true });
-      return runValidationSchema(state.values, name)
-        .then((x: any) => x)
-        .then((error: any) => {
-          dispatch({
-            type: 'SET_FIELD_ERROR',
-            payload: { field: name, value: error[name] },
-          });
-          dispatch({ type: 'SET_ISVALIDATING', payload: false });
+      return runValidationSchema(state.values, name).then((error: any) => {
+        dispatch({
+          type: 'SET_FIELD_ERROR',
+          payload: { field: name, value: error[name] },
         });
+        dispatch({ type: 'SET_ISVALIDATING', payload: false });
+      });
     }
 
     return Promise.resolve();
   });
 
-  const registerField = React.useCallback((name: string, { validate }: any) => {
-    fieldRegistry.current[name] = {
-      validate,
-    };
-  }, []);
+  const registerField = React.useCallback<FormikRegistration['registerField']>(
```

Comment:
please keep this a typescript only change. this change is actually breaking. not every field has a validate fn

Reply:
this was added bc validate could be undefined, and fieldRegistry only let me add string. If you prefer we can cast it as string and continue with this behavior

Reply:
rollback the any in the param so we dont have a logic change